cpp/thirdparty/download_dependencies.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This script downloads all the thirdparty dependencies as a series of tarballs
+# that can be used for offline builds, etc.
+
+set -e
+
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$#" -ne 1 ]; then
+  echo "Usage: $0 <destination-directory>"
+  exit
+fi
+
+_DST=$1
+
+BOOST_VERSION=1.67.0
```

Comment:
Is there a way we can avoid duplicating version numbers and download URLs between this and the CMake files?

Reply:
I don't care for the idea of maintaining the version numbers in two places either. I'm not sure how to get data from a file _in_ to CMake. I have done a bit of Google searching and I can't really find anything conclusive. CMake does have rudimentary string / regular expression parsing, that's probably the only way. I will take a crack at it and put a file like `thirdparty/versions.txt`